Digital Credentials: WKDigitalCredentialsPicker.h could use a cleanup

marcoscaceres opened the pull request:
#### 59f4befbba0472f64610ba4421306b2459f3e9cc
<pre>
Digital Credentials: WKDigitalCredentialsPicker.h could use a cleanup
<a href="https://bugs.webkit.org/show_bug.cgi?id=288007">https://bugs.webkit.org/show_bug.cgi?id=288007</a>
<a href="https://rdar.apple.com/145171561">rdar://145171561</a>

Reviewed by Aditya Keerthi.

Small clean up of WKDigitalCredentialsPicker.h and WKDigitalCredentialsPicker.mm.

* Source/WebKit/UIProcess/DigitalCredentials/WKDigitalCredentialsPicker.h:
* Source/WebKit/UIProcess/DigitalCredentials/WKDigitalCredentialsPicker.mm:

Canonical link: <a href="https://commits.webkit.org/290738@main">https://commits.webkit.org/290738@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/ba882509b57ce838afbca5945c9370fec6dbd975

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/90863 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/10400 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/45810 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/95884 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/41656 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/10795 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/18717 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/69861 "Passed tests") | [✅ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/27393 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/93864 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/8208 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/82368 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/50201 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/7978 "Passed tests") | | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/40778 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/78277 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/37827 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/97856 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/18058 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/122/builds/13260 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/78897 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/18318 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/78208 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/78080 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/19301 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/22558 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/119/builds/21175 "Passed tests") | [  ~~🛠 playstation~~](https://ews-build.webkit.org/#/builders/134/builds/11365 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/18067 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/23412 "Built successfully") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/17806 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/21262 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/19590 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

pxlcoder commented on Source/WebKit/UIProcess/DigitalCredentials/WKDigitalCredentialsPicker.h:
```diff
@@ -29,9 +29,8 @@
-#import <WebKit/WKWebView.h>
-#import <wtf/CompletionHandler.h>
 
+OBJC_CLASS WKWebView;
```
```suggestion
@class WKWebView;
```

`@class` is fine here since this header already uses Obj-C.

marcoscaceres replied:
makes sense. Thank you!

pxlcoder commented on Source/WebKit/UIProcess/DigitalCredentials/WKDigitalCredentialsPicker.mm:
```diff
@@ -38,7 +38,9 @@
 #import <WebCore/DigitalCredentialsResponseData.h>
 #import <WebCore/ExceptionData.h>
 #import <WebCore/IdentityCredentialProtocol.h>
+#import <WebKit/WKWebView.h>
```
Missed this earlier... you can just `#import "WKWebView.h"` since this code is in WebKit. And put it after `#import "Logging.h"` for sort order.

marcoscaceres replied:
ah, yeah. Will update now.

marcoscaceres committed:
Digital Credentials: WKDigitalCredentialsPicker.h could use a cleanup
https://bugs.webkit.org/show_bug.cgi?id=288007
rdar://145171561

Reviewed by Aditya Keerthi.

Small clean up of WKDigitalCredentialsPicker.h and WKDigitalCredentialsPicker.mm.

* Source/WebKit/UIProcess/DigitalCredentials/WKDigitalCredentialsPicker.h:
* Source/WebKit/UIProcess/DigitalCredentials/WKDigitalCredentialsPicker.mm:

Canonical link: https://commits.webkit.org/290738@main
